Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py

```diff
@@ -619,8 +628,15 @@ def _shard_by_directory(self, test_inputs, num_workers):
             tests_by_dir[directory].append(test_input)
 
         for directory, test_inputs in iteritems(tests_by_dir):
-            shard = TestShard(directory, test_inputs)
-            shards.append(shard)
+            shards_for_directory = 1 + len(test_inputs) // 256
```

Comment:
if we always use powers of two for shards_for_directory, then we should just be splitting existing shards, rather than rearranging everything as we reach a multiple of 256; see also my comment at https://github.com/WebKit/WebKit/pull/6148#issuecomment-1308984669